src/oaklib/implementations/bioportal/bioportal_implementation.py

```diff
@@ -1,22 +1,36 @@
-import requests
 import logging
 from dataclasses import dataclass, field
-from typing import Any, List, Dict, Union, Iterator, Iterable, Tuple
+from typing import Any, Dict, Iterable, Iterator, List, Tuple, Union
+from urllib.parse import quote
 
+import requests
 from oaklib.datamodels.text_annotator import TextAnnotation
 from oaklib.interfaces.basic_ontology_interface import PREFIX_MAP
-from oaklib.interfaces.search_interface import SearchInterface, SearchConfiguration
+from oaklib.interfaces.mapping_provider_interface import MappingProviderInterface
+from oaklib.interfaces.search_interface import SearchConfiguration, SearchInterface
 from oaklib.interfaces.text_annotator_interface import TextAnnotatorInterface
 from oaklib.types import CURIE
 from oaklib.utilities.apikey_manager import get_apikey_value
+from sssom import Mapping
+from sssom.sssom_datamodel import MatchTypeEnum
 
 REST_URL = "http://data.bioontology.org"
 
 ANNOTATION = Dict[str, Any]
 
+# See: 
+#   https://www.bioontology.org/wiki/BioPortal_Mappings 
+#   https://github.com/agroportal/project-management/wiki/Mappings
+SOURCE_TO_PREDICATE = {
+    'CUI': 'skos:closeMatch',
+    'LOOM': 'skos:closeMatch',
+    'REST': 'skos:relatedMatch', # maybe??
```

Comment:
yeh it's not super clear which skos property to use (other than SAME_URI), I think this is good enough to get started!